game.cpp

```diff
@@ -1491,8 +1493,28 @@ void test_game() //!OCLINT tests may be many
     g.tick();
     assert(!has_food(g));
     assert(!has_any_player_food_collision(g));
+
   }
-#endif
+  #ifdef FIX_ISSUE_440
+  // #440: Food changes the color of the player
+  {
+    game g;
+    const food f = g.get_food()[0];
+    // Food must be of a different color than the player,
+    // else nothing happens to the color of the player
+    const auto food_color = f.get_color();
+    const auto player_color = g.get_player(0).get_color();
+    assert(food_color != player_color);
+
+    const auto color_before = player_color;
+
+    put_player_on_food(g.get_player(0), f);
+    g.tick();
+
+    const auto color_after = g.get_player(0).get_color();
+    assert(color_before != color_after);
+  }
+  #endif // FIX_ISSUE_440
```

Comment:
Thanks for the review, I will change `auto` to the explicit data type, as suggested.

Note that, however, the C++ Core Guidelines recommend: [ES.11: Use auto to avoid redundant repetition of type names](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto). But I understand if the team prefers explicit data types, I will oblige :-)